jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -259,6 +264,16 @@ private Span setTagAsObject(String key, Object value) {
     return this;
   }
 
+  /**
+   * uses {@link #setTagAsObject} to only insert tags that don't have special significance
+   */
+  private void sanitizeAndSetTags(Map<String, Object> tags) {
```

Comment:
>And we do not discard any tags; they may not be stored as tags explicitly, but they are certainly "stored" one way or another. Nothing is being "sanitized" here.

I agree that the data contained in the tags are stored one way or the other. 
The tags themselves are discarded and cannot be retrieved by `Span#getTags`. Generally, if something is not discarded, I expect something set using the setter to be retrievable using a getter. 

In the class, setting tags using `#setTag` does not follow this behavior. 

At any rate, I don't see the point of arguing about semantics here. I'll inline this code.